### ISSUE TITLE
Search sometimes stops working

### ISSUE THREAD

**Issue by thomaslevesque** [microsoft/vscode#232178]
Sometimes, after using Search rather intensively for a while, it just stops working. To be exact, it seems to be doing _something_ when I try to search something new, but the results view is not updated and still show the results of the previous search.

I found this error in the Window output. It's repeated every time I try to search again.

```
2024-10-24 13:57:20.675 [error] TreeError [SearchView] Data tree node not found: [object Object]: Error: TreeError [SearchView] Data tree node not found: [object Object]
    at c$.D (vscode-file://vscode-app/c:/Users/tlevesque/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:405:34245)
    at c$.C (vscode-file://vscode-app/c:/Users/tlevesque/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:405:31263)
    at c$.updateChildren (vscode-file://vscode-app/c:/Users/tlevesque/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:405:31036)
    at vscode-file://vscode-app/c:/Users/tlevesque/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:2094:18478
    at Array.map (<anonymous>)
    at vscode-file://vscode-app/c:/Users/tlevesque/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:2094:18450
```

The only way I found to fix it is to restart VSCode

---

Does this issue occur when all extensions are disabled?: I don't know. I can't reproduce it on purpose, and I need the extensions to work, so I can't just disable the extensions until it happens (or doesn't).

- VS Code Version: 1.94.2
- OS Version: Windows 11 24H2

Steps to Reproduce: I don't know how to reproduce. It just happens sometimes when using the Search feature (to search in all files, not just the current file)

**Comment by thomaslevesque** [comment]
@scottforte I don't think it's related. It's not slow for me, even in a very large repo.

**Comment by andreamah** [comment]
@thomaslevesque does this still happen on the insiders build? I fixed some bugs in relation to https://github.com/microsoft/vscode/issues/231399

**Comment by scottforte** [comment]
fixed for me by dev reload then toggling "files to exclude icon" ![Image](https://github.com/user-attachments/assets/f0936f9c-96bc-4b56-98bc-0f0ac9602888)

**Comment by thomaslevesque** [comment]
> does this still happen on the insiders build? I fixed some bugs in relation to [#231399](https://github.com/microsoft/vscode/issues/231399)

@andreamah I can try the insiders build, but, the problem happens randomly and not very often (not even every day), so it will be a while before I can confirm whether it's fixed.

The bug you linked to seems to be exactly the same as my issue, though, so mine is a duplicate. I'll close it in favor of #231399
